src/backchannel_logout.rs

```diff
@@ -0,0 +1,469 @@
+    }
+
+    /// The `jti` claim. It's the unique identifier for this token and can be
+    /// used to detect replay attacks.
+    pub fn jti(&self) -> &str {
+        &self.jti
+    }
+
+    /// As per spec, a [`LogoutToken`] MUST either have the `sub`  or `sid`
+    /// claim and MAY contain both. You can match the [`Identifier`] to detect
+    /// which claims are present.
+    pub fn identifier(&self) -> &Identifier {
+        &self.identifier
+    }
+
+    /// A [`LogoutToken`] is compatible with the [SET standard from RFC 8417][1]
+    ///
+    /// [1]: <https://www.rfc-editor.org/info/rfc8417>
+    pub fn events(&self) -> &HashMap<String, serde_json::Value> {
+        &self.events
+    }
+}
+impl<'de> Deserialize<'de> for LogoutToken {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value: serde_json::Value = serde_json::Value::deserialize(deserializer)?;
+        if let serde_json::Value::Object(ref map) = value {
+            if map.contains_key("nonce") {
+                return Err(<D::Error as Error>::custom("nonce claim is prohibited"));
+            }
+        }
+
+        #[derive(Deserialize)]
+        struct Repr {
+            /// The issuer of this token
+            iss: IssuerUrl,
+            /// The audience this token is intended for
+            #[serde(deserialize_with = "deserialize_string_or_vec")]
+            aud: Vec<Audience>,
+            /// Time at which this token was issued
+            #[serde(with = "serde_utc_seconds")]
+            iat: DateTime<Utc>,
+            /// The unique identifier for this token. This can be used to detect
+            /// replay attacks.
+            jti: String,
+            #[serde(flatten)]
+            identifier: Identifier,
+            events: HashMap<String, serde_json::Value>,
+        }
+
+        let token: Repr = serde_json::from_value(value).map_err(<D::Error as Error>::custom)?;
+
+        token
+            .events
+            // according to the spec, this event must be included in the mapping
+            .get("http://schemas.openid.net/event/backchannel-logout")
+            .ok_or_else(|| {
+                <D::Error as Error>::custom("token is missing correct JSON Object in events claim")
+            })?
+            // and it must be a JSON object and MAY BE empty but is allowed to
+            // contain fields
+            .as_object()
+            .ok_or_else(|| <D::Error as Error>::custom("not a JSON Object"))?;
+        Ok(LogoutToken {
+            iss: token.iss,
+            aud: token.aud,
+            iat: token.iat,
+            jti: token.jti,
+            identifier: token.identifier,
+            events: token.events,
+        })
+    }
+}
+
+/// A [`LogoutToken`] MUST contain either a `sub` or a `sid` claim and MAY
+/// contain both. This enum represents these three possibilities.
+#[derive(Debug, Hash, Clone, PartialEq, Eq)]
+pub enum Identifier {
```

Comment:
I like this enum since it ensures that we can't represent a token with both fields set to `None`! I would just suggest naming it something a bit more descriptive like `LogoutIdentifier`.